evaluator/src/main/kotlin/ProjectSourceRule.kt

```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright (C) 2022 EPAM Systems, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ * License-Filename: LICENSE
+ */
+
+package org.ossreviewtoolkit.evaluator
+
+import java.io.File
+
+/**
+ * A [OrtResultRule] which allows downloading the project's source code if needed.
+ */
+open class ProjectSourceRule(
+    ruleSet: RuleSet,
+    name: String,
+    projectSourceTreeResolver: SourceTreeResolver = SourceTreeResolver.forRemoteRepository(
+        ruleSet.ortResult.repository.vcsProcessed
+    )
+) : OrtResultRule(ruleSet, name) {
+    /**
+     * The directory containing the source code of the project. Accessing the property for the first time triggers a
+     * clone and may take a while.
+     */
+    @Suppress("UnusedPrivateMember")
```

Comment:
This should be `@Suppress("unused")` instead. But why are you introducing the property in this commit already if it's unused?

Reply:
BTW, even with the later commits, it looks like this property indeed can be private.

Reply:
The property is public intentionally, so that user can used it to write custom matchers.